client/config/toml.go

```diff
@@ -64,7 +64,8 @@ func writeConfigFile(configFilePath string, config interface{}) error {
 }
 
 // getClientConfig reads values from client.toml file and unmarshalls them into ClientConfig
-func getClientConfig(configPath string, v *viper.Viper) (*Config, error) {
+func getClientConfig(configPath string) (*Config, error) {
+	v := viper.New()
```

Comment:
You should continue to use the provided viper. Creating a new one erases any thing that one might need to set up for viper before this (e.g. env var prefix). It also overwrites the global viper instance which can cause problems too (some stuff uses that global viper instance).

Reply:
The provided viper instance checks the wrong file due to AutoCLI initially giving it the default home.

Reply:
It will then see the `client.toml` in the incorrect directory first and go with that one.

Reply:
> It also overwrites the global viper instance which can cause problems too

This doesn't set the viper ref that was passed in before. In fact, the function does not currently accept a ref.

Reply:
After more investigation. I see what you mean. A better fix would be recreating the viper instance in root.go on a per app basis (after the root command has already been initialized).
I'll post changes tomorrow.